lua/core/hook.lua

```diff
@@ -36,6 +36,7 @@ local hooks = {
   script_pre_init = Hook.new(),
   script_post_init = Hook.new(),
   script_post_cleanup = Hook.new(),
+  system_pre_device_scan = Hook.new(),
```

Comment:
In hindsight maybe this hook isn’t necessary? I could just call serial.handler in the global scope of the mod since mods are loaded as the last step of startup script.

Reply:
This might be worth discussing again before merge. On the one hand it works as it is now. But on the other, I am of the opinion that fewer hooks is better. Another point is that the device scan stage isn't referenced very often from what I know of so it might come across as a bit mysterious since it looks different from the other hooks.

A `system_pre_startup` hook would fit the existing hooks a little more closely, but would require moving the mod/hook initialization outside of `startup.lua`.

Reply:
Good points. I'm in agreement and inclined to remove it. I was thinking maybe it makes sense to have an official hook to decouple serial registration from mod loading, even if they happen at the same time right now.

A good future addition might be to rescan devices when a new handler is added. Then it wouldn't matter if the handlers are registered in a mod or a script.